simple/msg_setname.c

```diff
@@ -0,0 +1,647 @@
+		do {
+			ret = fi_cq_read(rcq, &comp, 1);
+			if (ret < 0 && ret != -FI_EAGAIN) {
+				FT_PRINTERR("fi_cq_read", ret);
+				return ret;
+			}
+		} while (ret == -FI_EAGAIN);
+
+		fprintf(stdout, "Received data from client: %s\n", (char *)buf);
+	}
+
+	return 0;
+}
+
+static int setup_handle(void)
+{
+	static char buf[BUFSIZ];
+	struct addrinfo *ai, aihints;
+	struct fi_info *fi;
+	int ret;
+
+	memset(&aihints, 0, sizeof aihints);
+	aihints.ai_flags = AI_PASSIVE;
+	ret = getaddrinfo(opts.src_addr, opts.src_port, &aihints, &ai);
+	if (ret == EAI_SYSTEM) {
+		FT_PRINTERR("getaddrinfo", -ret);
+		return -ret;
+	} else if (ret) {
+		FT_ERR("getaddrinfo: %s\n", gai_strerror(ret));
+		return -FI_ENODATA;
+	}
+
+	switch (ai->ai_family) {
+	case AF_INET:
+		hints->addr_format = FI_SOCKADDR_IN;
+		break;
+	case AF_INET6:
+		hints->addr_format = FI_SOCKADDR_IN6;
+		break;
+	case AF_IB:
+		hints->addr_format = FI_SOCKADDR_IB;
+		break;
+	}
+
+	/* Get fabric info */
+	ret = fi_getinfo(FT_FIVERSION, NULL, NULL, 0, hints, &fi);
+	if (ret) {
+		FT_PRINTERR("fi_getinfo", ret);
+		goto free_ai;
+	}
+
+	/* Open the fabric */
+	ret = fi_fabric(fi->fabric_attr, &fab, NULL);
+	if (ret) {
+		FT_PRINTERR("fi_fabric", ret);
+		goto free_fi;
+	}
+
+	/* Open a passive endpoint */
+	ret = fi_passive_ep(fab, fi, &pep, NULL);
+	if (ret) {
+		FT_PRINTERR("fi_passive_ep", ret);
+		goto free_fabric;
+	}
+
+	ret = fi_setname(&pep->fid, ai->ai_addr, ai->ai_addrlen);
+	if (ret) {
+		FT_PRINTERR("fi_setname", ret);
+		goto free_pep;
+	}
+
+	ret = fi_getname(&pep->fid, bound_addr, &bound_addr_len);
+	if (ret) {
+		FT_PRINTERR("fi_getname", ret);
+		goto free_pep;
+	} else if (bound_addr->sa_family != AF_INET) {
+		FT_ERR("got unexpected address family %d\n",
+				bound_addr->sa_family);
+		/* otherwise ignore this error */
+	} else if (((struct sockaddr_in *)bound_addr)->sin_port == 0) {
+		FT_ERR("port number is 0 after fi_setname\n");
+		ret = -FI_EINVAL;
+		goto free_pep;
+	}
+	printf("bound_addr: \"%s\"\n",
+			sockaddrstr(bound_addr, bound_addr_len, buf, BUFSIZ));
+
+	hints->handle = &pep->fid;
+	goto free_fi;
+
+free_pep:
+	fi_close(&pep->fid);
+free_fabric:
+	fi_close(&fab->fid);
+
+free_fi:
+	fi_freeinfo(fi);
+free_ai:
+	freeaddrinfo(ai);
+	return ret;
+}
+
+
+int main(int argc, char **argv)
+{
+	int op, ret;
+	opts = INIT_OPTS;
+
+	hints = fi_allocinfo();
+	if (!hints)
+		return EXIT_FAILURE;
+
+	while ((op = getopt(argc, argv, "h" ADDR_OPTS INFO_OPTS)) != -1) {
+		switch (op) {					
+		default:
+			ft_parse_addr_opts(op, optarg, &opts);
+			ft_parseinfo(op, optarg, hints);
+			break;
+		case '?':
+		case 'h':
+			ft_usage(argv[0], "A simple MSG client-sever example.");
+			return EXIT_FAILURE;
+		}
+	}
+		
+	if (optind < argc)
+		opts.dst_addr = argv[optind];
+
+	hints->ep_attr->type	= FI_EP_MSG;
+	hints->caps		= FI_MSG;
+	hints->mode		= FI_LOCAL_MR;
+	hints->addr_format	= FI_SOCKADDR;
+
+	/* Fabric and connection setup */
+	if (opts.src_addr || opts.src_port) {
```

Comment:
What is the purpose of this conditional? Should this only be run for the server? Both server and client will have `opts.src_port` set since they both use `INIT_OPTS` for initialization.